tests/job/test_factory.py

```diff
@@ -24,3 +39,24 @@ def test_core(self):
 
     def test_base(self):
         self.assertGreater(len(self.factory._job_class_dict), 0)
+
+    def test_call_standard(self):
+        job = self.factory('ScriptJob', 'foo.bar')
+        self.assertIsInstance(job, ScriptJob, msg=f"Got a {type(job)} instead of a ScriptJob")
+        self.assertEqual('foo_bar', job.name, msg=f"Job name failed to set, expected foo but got {job.name}")
```

Comment:
This test would need to be adapted with my above request ('.' would be converted to 'd' not to '_').